addons/website_slides/static/src/js/slides_course_page.js

```diff
@@ -0,0 +1,148 @@
+ * as method to mark the slide as completed / uncompleted.
+ */
+export const SlideCoursePage = publicWidget.Widget.extend({
+    events: {
+        'click button.o_wslides_button_complete': '_onClickComplete',
+    },
+
+    custom_events: {
+        'slide_completed': '_onSlideCompleted',
+        'slide_to_complete': '_onSlideToComplete',
+    },
+
+    xmlDependencies: ['/website_slides/static/src/xml/website_slides_sidebar.xml'],
+
+    /**
+     * Greens up the bullet when the slide is completed
+     *
+     * @public
+     * @param {Integer} slideId
+     * @param {Boolean} completed
+     */
+    toggleCompletionButton: function (slide, completed = true) {
+        const $button = this.$(`.o_wslides_sidebar_done_button[data-id=${slide.id}]`);
+
+        const newButton = QWeb.render('website.slides.sidebar.done.button', {
+            slideId: slide.id,
+            slideCompleted: completed,
+            canSelfMarkUncompleted: slide.canSelfMarkUncompleted,
+            canSelfMarkCompleted: slide.canSelfMarkCompleted,
+            isMember: slide.isMember,
+        });
+
+        if ($button.length) {
+            $button.replaceWith(newButton);
+            const $elem = this.$(`.o_wslides_sidebar_done_button[data-id="${slide.id}"]`);
+            $elem.data('completed', completed);
+        }
+    },
+
+    /**
+     * Updates the progressbar whenever a lesson is completed
+     *
+     * @public
+     * @param {*} channelCompletion
+     */
+    updateProgressbar: function (channelCompletion) {
+        const completion = Math.min(100, channelCompletion);
+
+        const $completed = $('.o_wslides_channel_completion_completed');
+        const $progressbar = $('.o_wslides_channel_completion_progressbar');
+
+        if (completion < 100) {
+            // Hide the "Completed" text and show the progress bar
+            $completed.addClass('d-none');
+            $progressbar.removeClass('d-none').addClass('d-flex');
+        } else {
+            // Hide the progress bar and show the "Completed" text
+            $completed.removeClass('d-none');
+            $progressbar.addClass('d-none').removeClass('d-flex');
+        }
+
+        $progressbar.find('.progress-bar').css('width', `${completion}%`);
+        $progressbar.find('.o_wslides_progress_percentage').text(completion);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Once the completion conditions are filled,
+     * rpc call to set the relation between the slide and the user as "completed"
+     *
+     * @private
+     * @param {Integer} slideId: the id of slide to set as completed
+     * @param {Boolean} completed: true to mark the slide as completed
+     *     false to mark the slide as not completed
+     */
+    _toggleSlideCompleted: async function (slide, completed = true) {
```

Comment:
Same question here, do we have to pass the slide? Would be nice to use "_getSlide".